src/libraries/Hooks.sol

```diff
@@ -96,10 +96,12 @@ library Hooks {
 
     /// @notice performs a hook call using the given calldata on the given hook
     function callHook(IHooks self, bytes memory data) internal {
-        (bytes4 expectedSelector, bytes4 selector) = _callHook(self, data);
+        if (msg.sender != address(self)) {
```

Comment:
ooh this is an interesting line of code.. address(self) reads like address(this), but is actually checking address(hook)... I feel like I lean towards a rename of `callHook(IHooks hook, bytes memory data)`

Reply:
could collapse the if statement by doing 
if (msg.sender == address(hook) return;
_callHook()
if(selector != expectedSelector) revert .. 

which may be clean? but try it out and see if you like it.. not that opinionated

Reply:
I think I like that! Honestly it took me a while to figure out what was what so I think this will make it more clear